objectivec/GPBWellKnownTypes.m

```diff
@@ -115,3 +115,40 @@ - (void)setTimeIntervalSince1970:(NSTimeInterval)timeIntervalSince1970 {
 }
 
 @end
+
+NSString *const GPBTypeGoogleApisComPrefix = @"type.googleapis.com/";
+
+@implementation GPBAny (GBPWellKnownTypes)
+
+- (NSString*)getAnyTypeName {
```

Comment:
if message hasn't been set, then the typeURL will be nil. So calling typeName on it will return nil.

You can keep the assert if you want. Please change the name of the method to

- (NSString*)typeName